src/Paths/Directory.ts

```diff
@@ -28,7 +26,7 @@ export const root = "" as Directory;
 const invalidCharacters = ["/", "*", "?", "[", "]", "!", "\\", "~", "|", "#", '"', "'", " "];
 
 /** A valid character is any character that is not one of the invalid characters */
-export const oneValidCharacter = `[^${escapeRegExp(invalidCharacters.join(""))}]`;
+export const oneValidCharacter = `[^${invalidCharacters.map((c) => "\\" + c).join("")}]`;
```

Comment:
To easily import the path manipulation function into the worker, there can't be any imports of modules that would go into the vendor bundle. Thus, I just manually escape the characters here.

Reply:
This needs to be commented somewhere, because the restriction is highly non-obvious. Is this for space reasons, or functional reasons? If it's not something that will cause the build to break, it *will* need to be enforced with a test, otherwise this constraint will be violated very quickly.

Reply:
I guess the comment in `worker.js` got lost in one of the iterations.

The issue is that `worker.ts` needs to *synchronously* define `customTSWorkerFactory`. If it (recursively) imports anything stuff that gets put into `vendor.bundle.js` (i.e. from `node_modules), then it gets defined asynchously.

I'm not sure how to get it to fail the build, but it errors out as soon as you open the editor.

Another option would be to write much messier code to do the imports in `worker.js` (so the imports are not at the top-level), rather than impose this restriction.

Reply:
It looks like this is a bug in webpack (https://github.com/webpack/webpack/issues/18615), though we'll need to work around it for the moment.

Reply:
OK, so there's two major things that need doing here: 
1. The situation needs to be documented. An explanation of what's going on with synchronous loading in monaco with workers and a link to that bug, at minimum.
2. Either a test needs to be added, or you need to do the complicated thing. "We have to manually check to see if the editor still works on each change" is not an acceptable state of affairs.